src/game/common/system/mempoolobj.h

```diff
@@ -76,45 +76,44 @@ private: \
     virtual MemoryPool *Get_Object_Pool() override { return Get_Class_Pool(); } \
 \
 public: \
-    enum classname##MagicEnum{ classname##_GLUE_NOT_IMPLEMENTED }; \
+    enum classname##MagicEnum{ classname##_GLUE_NOT_IMPLEMENTED = 0 }; \
 \
-    void *operator new(size_t size); \
-    void *operator new(size_t size, classname##MagicEnum) \
+    void *operator new(size_t size, classname##MagicEnum = classname##_GLUE_NOT_IMPLEMENTED) \
     { \
         captainslog_dbgassert(size == sizeof(classname), \
             "The wrong operator new is being called; ensure all objects in the hierarchy have MemoryPoolGlue set up " \
             "correctly"); \
         return Get_Class_Pool()->Allocate_Block(); \
     } \
-    void operator delete(void *ptr) \
+    void operator delete(void *ptr) = delete; \
+    void operator delete(void *ptr, classname##MagicEnum) \
     { \
-        captainslog_debug("Please call deleteInstance instead of delete."); \
+        captainslog_dbgassert(0, "Please call Delete_Instance instead of delete."); \
         Get_Class_Pool()->Free_Block(ptr); \
-    } \
-    void operator delete(void *ptr, classname##MagicEnum) { Get_Class_Pool()->Free_Block(ptr); }
+    }
+
+/* NOTE: Operator delete above must exist because of the pairing operator new. However, it should never be called directly.
+ * Reason being, a pointer to a derived class of this class, must use the correct memory pool for deletion. This is possible
+ * by calling Get_Object_Pool function. However, operator delete is called after class destructor, and class destructor
+ * will destroy virtual table. Therefore operator delete cannot be used. */
 
 #define IMPLEMENT_POOL(classname) IMPLEMENT_NAMED_POOL(classname, classname);
 
 #define IMPLEMENT_ABSTRACT_POOL(classname) \
 private: \
     virtual MemoryPool *Get_Object_Pool() override { throw CODE_01; } \
-\
-    enum classname##MagicEnum{ classname##_GLUE_NOT_IMPLEMENTED }; \
-\
-protected: \
-    void *operator new(size_t size); \
-    void *operator new(size_t size, classname##MagicEnum); \
-    void operator delete(void *ptr) { captainslog_debug("this should be impossible to call (abstract base class)"); } \
```

Comment:
I have seen this error message in the decompile, but it's useless to keep. Best to just delete the function and let the compiler give an error straight.